sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/UpdatingSpecificationFilesGenerator.java

```diff
@@ -0,0 +1,253 @@
+    Path checkTemplatePath = Path.of("src", "test", "resources", "templates", "GenericCheckTemplate.java");
+    try {
+      Files.copy(checkTemplatePath, checkPath);
+
+      String content = Files.readString(checkPath, charset);
+      content = content.replaceAll("GenericCheckTemplate", checkName);
+      content = content.replaceAll("<RSPEC-KEY>", rspecKey);
+      Files.write(checkPath, content.getBytes(charset));
+    } catch (IOException e) {
+      LOG.error("Error while writing Check file with name \"" + checkName + ".java\", please fix manually", e);
+      throw new RuntimeException(e);
+    }
+
+    String successMessage = String.format("Successfully generated Check \"%s.java\" with rspecKey %s", checkName, rspecKey);
+    LOG.info(successMessage);
+  }
+
+  public void writeCheckTestFile(String checkName) {
+    Path checkTestPath = Path.of("src", "test", "java", "org", "sonar", "plugins", "secrets", "checks", checkName + "Test.java");
+    Path checkTestTemplatePath = Path.of("src", "test", "resources", "templates", "GenericCheckTestTemplate.java");
+    try {
+      Files.copy(checkTestTemplatePath, checkTestPath);
+
+      String content = Files.readString(checkTestPath, charset);
+      content = content.replaceAll("GenericCheckTemplate", checkName);
+      content = content.replaceAll("GenericCheckTemplateTest", checkName + "Test");
+      Files.write(checkTestPath, content.getBytes(charset));
+    } catch (IOException e) {
+      LOG.error("Error while writing Check Test file with name \"" + checkName + "Test.java\", please fix manually", e);
+      throw new RuntimeException(e);
+    }
+
+    String successMessage = String.format("Successfully generated Check \"%sTest.java\"", checkName);
+    LOG.info(successMessage);
+  }
+
+  public void writeSpecificationFileDefinition() {
+    Path specificationsDirectoy = Path.of("src", "main", "resources", "org", "sonar", "plugins", "secrets", "configuration");
+    String[] extensionsToSearchFor = new String[] {"yaml"};
+    Collection<File> files = FileUtils.listFiles(new File(specificationsDirectoy.toUri()), extensionsToSearchFor, false);
+
+    List<String> listOfFileNames = files.stream().map(File::getName).sorted().collect(Collectors.toList());
+
+    Path specificationDefinitionPath = Path.of("src", "main", "java", "org", "sonar", "plugins", "secrets", "SecretsSpecificationFilesDefinition.java");
+    Path specificationDefinitionTemplatePath = Path.of("src", "test", "resources", "templates", "SecretsSpecificationFilesDefinitionTemplate.java");
+
+    try {
+      Files.copy(specificationDefinitionTemplatePath, specificationDefinitionPath, StandardCopyOption.REPLACE_EXISTING);
+
+      String content = Files.readString(specificationDefinitionPath, charset);
+      content = content.replaceAll("<REPLACE-WITH-SET-OF-FILENAMES>", generateSpecificationDefinitionSet(listOfFileNames));
```

Comment:
Why `replaceAll` instead of `replace`? The first argument is not a regex, so `replace` will be faster.